updater/config.gni

```diff
@@ -14,7 +14,19 @@ declare_args() {
   brave_enable_updater = false
 }
 
+declare_args() {
+  # It would be nice to declare build_omaha4 in the declare_args block above.
+  # But its default value depends on brave_enable_updater. In this case, GN
+  # requires us to create a separate declare_args block.
+  build_omaha4 = brave_enable_updater && !is_official_build
+}
+
 if (brave_enable_updater) {
   assert(target_os == "mac",
          "brave_enable_updater currently only takes effect on macOS.")
 }
+
+if (build_omaha4) {
+  assert(brave_enable_updater,
```

Comment:
maybe have one more assert here: `!notarize || (notarize && is_official_build)`?

Reply:
I have a suspicion I understand the motivation for the additional assert, but am not 100% sure.

Before this PR, signing the browser with O4 built from source produced a binary that actually did not meet codesigning requirements. Submitting this binary to notarization (esp. when notarizing the PKG) then lead to an error from Apple's remote server.

The reason for this is that upstream's code is not meant to sign a browser that contains O4 built from source. Upstream's signing code only works if O4 in the browser is already signed.

Is the motivation for the assert to catch this case? If yes, then I don't believe `notarize` and `is_official_build` fully capture the above situation. (Problems already start with signing, before notarization, and if `is_official_build` is used as a proxy to determine whether the browser vs. just `sign_updater` is being built, then I think it's a poor proxy. Otherwise, I don't understand the motivation for `is_official_build` in the `assert`.)

But in any case - if the motivation for the assert is to catch the situation I described - then I'm not sure it needs to be there. Yes, at the moment, upstream's code does not support signing the browser with O4 built from source. But maybe this will change in the future. Then, it will make no sense to have the `assert` in our code.

(As a side note, I believe `!notarize || (notarize && is_official_build)` is logically equivalent to `!notarize || is_official_build`.)

Reply:
> Do we need to notarize non-release binaries?

It can be useful during development. I have been notarizing a lot in the past two weeks.